crates/nu-command/src/filesystem/open.rs

```diff
@@ -91,6 +91,19 @@ impl Command for Open {
                         file: path.item.to_string(),
                         span,
                     },
+                    ShellError::GenericError {
+                        error,
+                        msg,
+                        span: _,
+                        help,
+                        inner,
+                    } if error.as_str() == "Permission denied" => ShellError::GenericError {
+                        error,
+                        msg,
+                        span: Some(arg_span),
+                        help,
+                        inner,
+                    },
```

Comment:
Could you add a comment explaining why this is here? It makes sense in the context of the PR, but I think this might be a bit confusing if you were to stumble across it.